src/seedu/addressbook/data/person/Address.java

```
@@ -27,15 +27,36 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        partitionAddress();
     }
-
     /**
      * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     *  Partitions address into its block, street, unit and postal code.
+     *
+     *  Assumes valid input.
+     */
+    private void partitionAddress() {
+        try {
+            String[] categories = value.split(",\\s", 4);
+        Block block = new Block(Integer.parseInt(categories[0]));
+        Street street = new Street(categories[1]);
+        Unit unit = new Unit(categories[2]);
+        PostalCode postalCode = new PostalCode(Integer.parseInt(categories[3]));
+        }
+        catch(ArrayIndexOutOfBoundsException e) {
+
+        }
+        catch(Exception exception){
+
+        }
```

You use try-catch with empty catch blocks, I don't think they're useful.